app/dashboard/tasks.py

```diff
@@ -330,3 +332,69 @@ def recalculate_earning(self, pk, retry: bool = True) -> None:
     earning = Earning.objects.get(pk=pk)
     src = earning.source
     src.save()
+
+
+@app.shared_task(bind=True, max_retries=1)
+def record_visit(self, user_pk, profile_pk, ip_address, visitorId, useragent, referrer, path, session_key, utm, retry: bool = True) -> None:
+    """
+    :param self: Self
+    :param user_pk: user primary
+    :param profile_pk: profile primary
+    :param ip_address: get_ip(request)
+    :param visitorId: request.COOKIES.get("visitorId", None)
+    :param useragent: request.META['HTTP_USER_AGENT']
+    :param referrer: request.META.get('HTTP_REFERER', None)
+    :param path: request.META.get('PATH_INFO', None)
+    :param session_key: request.session._session_key
+    :param utm: _get_utm_from_cookie(request)
+    :return: None
+    """
+
+    from django.contrib.auth.models import User
+    user = User.objects.filter(pk=user_pk).first() if user_pk else None
+    profile = Profile.objects.filter(pk=profile_pk).first() if profile_pk else None
+    if user and profile:
+        try:
+            profile.last_visit = timezone.now()
+            profile.save()
+        except Exception as e:
+            logger.exception(e)
+        try:
+            profile_dict.delay(profile.pk)
+        except Exception as e:
+            logger.exception(e)
+        try:
```

Comment:
Se can remove L363-L366

Reply:
We can likely remove the `try...except` from `profile_dict.delay(profile.pk)` - this was copied over from `context.py` as is, but lets keep the recalc here so that it's triggered regularly 👍